### PR TITLE
L1T phase-2: enable tkGlbMuons

### PR DESCRIPTION
**Pull request by folguera**
#### PR description:

This PR resurrects the tkGlbMuon collection for the HLT TDR. This PR superseeds #862. 

#### PR validation:

ongoing, if we are convinced that this works, we will want this merged too.

### CODE REVIEW

**Comment by folguera** [comment]
The code has been validated by the group and it is featuring the expected results, hence this is ready to be merged from our side.

**Comment by folguera** [comment]
One caveat, recently pointed out by Cristina that has to be said, we are double checking efficiencies also with the ntuple trees. We will have results before the end of the week.

**Comment by folguera** [comment]
@rekovic this PR now is fully validated, both efficiencies and rates have been studied. This is ready to be merged.

File: DataFormats/L1TCorrelator/interface/TkMuon.h [hunk 1]
```diff
@@ -58,6 +65,7 @@ namespace l1t {
   private:
     // used for the Naive producer
     edm::Ref<l1t::RegionalMuonCandBxCollection> muRef_;
+    edm::Ref<l1t::EMTFTrackCollection> emtfTrk_;
```
Comment (rekovic) [inline DataFormats/L1TCorrelator/interface/TkMuon.h:68]: This leads to crash in OutputModule due  to missing streamer of `EMTFTrack`

```
----- Begin Fatal Exception 14-Dec-2020 18:51:55 CET-----------------------
An exception of category 'FatalRootError' occurred while
   [0] Processing  Event run: 1 lumi: 1140 event: 189079 stream: 6
   [1] Running path 'FEVTDEBUGHLToutput_step'
   [2] Calling method for module PoolOutputModule/'FEVTDEBUGHLToutput'
   Additional Info:
      [a] Fatal Root Error: @SUB=TStreamerInfo::Build
l1t::TkMuon: edm::Ref<vector<l1t::EMTFTrack>,l1t::EMTFTrack,edm::refhelper::FindUsingAdvance<vector<l1t::EMTFTrack>,l1t::EMTFTrack> > has no streamer or dictionary, data member "emtfTrk_" will not be saved
```

**Comment by rekovic** [comment]
@folguera Thanks.
As soon as AFS starts behaving will test and integrate

**Pull request merged**